File: SampleApp/Objective-C/SampleApp/AdMob/Banner/AdMobBannerViewController.m
//
//  AdMobBannerViewController.m
//  SampleApp
//
//  Copyright © 2020 fluct, inc. All rights reserved.
//

#import "AdMobBannerViewController.h"
#import <GoogleMobileAds/GoogleMobileAds.h>

@interface AdMobBannerViewController () <GADBannerViewDelegate>
@property (nonatomic, nullable) GADBannerView *bannerView;
@end

// AdUnitIDを適切なものに変えてください
static NSString *const kAdUnitID = @"ca-app-pub-3010029359415397/1722697861";

@implementation AdMobBannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.

    GADBannerView *bannerView = [[GADBannerView alloc] initWithAdSize:GADAdSizeBanner];
    bannerView.rootViewController = self;
    bannerView.adUnitID = kAdUnitID;

    [self.view addSubview:bannerView];
    self.bannerView = bannerView;
    [bannerView loadRequest:[GADRequest request]];
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    self.bannerView.center = self.view.center;
}

#pragma mark - GADBannerViewDelegate

- (void)bannerViewDidReceiveAd:(GADBannerView *)bannerView {
    NSLog(@"%s", __FUNCTION__);
}

- (void)bannerViewWillPresentScreen:(GADBannerView *)bannerView {
    NSLog(@"%s", __FUNCTION__);
}

- (void)bannerViewDidDismissScreen:(GADBannerView *)bannerView {
    NSLog(@"%s", __FUNCTION__);
}

- (void)bannerView:(GADBannerView *)bannerView didFailToReceiveAdWithError:(NSError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
}

@end

File: SampleApp/Swift/SampleApp/AdMob/Banner/AdMobBannerViewController.swift
//
//  AdMobBannerViewController.swift
//  SampleApp
//
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

import UIKit
import GoogleMobileAds

// AdUnitIDを適切なものに変えてください
private let adUnitID: String = "ca-app-pub-3010029359415397/1722697861"
// AdSizeを適切なものに変えてください
private let adSize: GADAdSize = kGADAdSizeBanner

class AdMobBannerViewController: UIViewController {

    private var bannerView: GADBannerView?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.

        let bannerView = GADBannerView(adSize: kGADAdSizeBanner)
        bannerView.rootViewController = self
        bannerView.adUnitID = adUnitID

        view.addSubview(bannerView)
        self.bannerView = bannerView
        bannerView.load(GADRequest())
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        bannerView?.center = view.center
    }

}

// MARK: - GADBannerViewDelegate

extension AdMobBannerViewController: GADBannerViewDelegate {

    func bannerViewDidReceiveAd(_ bannerView: GADBannerView) {
        print(#function)
    }

    func bannerViewWillPresentScreen(_ bannerView: GADBannerView) {
        print(#function)
    }

    func bannerViewWillDismissScreen(_ bannerView: GADBannerView) {
        print(#function)
    }

    func bannerViewDidDismissScreen(_ bannerView: GADBannerView) {
        print(#function)
    }

    func bannerView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: Error) {
        print(#function, error)
    }

}
